src/functions/getFirstExistingParentPath.ts

```diff
@@ -8,15 +8,19 @@ import Dependencies from '@/src/types/dependencies'
  * @param dependencies - Dependencies container
  */
 async function getFirstExistingParentPath(directoryPath: string, dependencies: Dependencies): Promise<string> {
-	let parentDirectoryPath = directoryPath
+	let parentDirectoryPath = dependencies.pathNormalize(directoryPath)
 	let parentDirectoryFound = await isDirectoryExisting(parentDirectoryPath, dependencies)
 
-	while (!parentDirectoryFound) {
-		parentDirectoryPath = dependencies.pathNormalize(parentDirectoryPath + '/..')
+	for (let i = 0; i < 4096 && !parentDirectoryFound; ++i) {
+		const newParentDirectoryPath = dependencies.pathNormalize(parentDirectoryPath + '/..')
+		if (parentDirectoryPath === newParentDirectoryPath) {
+			return ''
```

Comment:
What does it mean if this function returns an empty string?

Reply:
it failed basically. I left this behavior the same as is used in heroic. Personally I'd probably rather do `undefined` but both evaluate to false and you can't have an empty string as a directory so I suppose it works

Reply:
added a constant for this!